packages/agent/src/store-data.ts

```diff
@@ -158,8 +158,8 @@ export class DwnDataStore<TStoreObject extends Record<string, any> = Jwk> implem
 
       // set the recordId in the messageParams to update the existing record
       // set the dateCreated to the existing dateCreated as this is an immutable property
-      messageParams.recordId = matchingRecordEntry.recordsWrite?.recordId;
-      messageParams.dateCreated = matchingRecordEntry.recordsWrite?.descriptor.dateCreated;
+      messageParams.recordId = matchingRecordEntry.recordsWrite!.recordId;
+      messageParams.dateCreated = matchingRecordEntry.recordsWrite!.descriptor.dateCreated;
```

Comment:
not important but you could hide the complexity in a type guard and not have to do the `!`